doc: oauth middleware pathFilter (#4745)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-middleware/middleware-oauth2.md

      type: middleware.http.oauth2
```

> ## Path Filtering for Least Privilege Security

Reviewer: ```suggestion
## Request path filtering
```

> The `pathFilter` field allows you to apply OAuth2 authentication selectively based on request paths using regex patterns. This enables implementing the least privilege principle by configuring multiple OAuth2 middlewares with different scopes for different API endpoints, ensuring users only receive the minimum permissions necessary for their intended operations.

Reviewer: ```suggestion
The `pathFilter` field allows you to selectively apply OAuth2 authentication based on the HTTP request path using a regex pattern. This enables scenarios such as configuring multiple OAuth2 middlewares with different scopes for different API endpoints, implementing the least privilege principle by ensuring users only receive the minimum permissions necessary for their intended operation.
```

> ### Example: Separate User and Admin Access

Reviewer: ```suggestion
### Example: Separate read-only and admin user access
```

> 

Reviewer: ```suggestion
In the following configuration:
- Requests to `/api/users/*` endpoints receive tokens with a read-only user scopes
- Requests to `/api/admin/*` endpoints receive tokens with full admin scopes

This reduces security risk by preventing unnecessary privilege access and limiting the blast radius of compromised tokens.
```

```yaml

> # Read-only user access middleware

Reviewer: ```suggestion
# User with read-only access scope
```

apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: oauth2-users
spec:
  type: middleware.http.oauth2
  version: v1
  metadata:
  - name: clientId
    value: "<your client ID>"
  - name: clientSecret
    value: "<your client secret>"
  - name: scopes
    value: "user:read profile:read"
  - name: authURL
    value: "https://accounts.google.com/o/oauth2/v2/auth"
  - name: tokenURL
    value: "https://accounts.google.com/o/oauth2/token"
  - name: redirectURL
    value: "http://myapp.com/callback"
  - name: pathFilter
    value: "^/api/users/.*"
---

> # Full admin access middleware  

Reviewer: ```suggestion
# User with full admin access scope
```

apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: oauth2-admin
spec:
  type: middleware.http.oauth2
  version: v1
  metadata:
  - name: clientId
    value: "<your client ID>"
  - name: clientSecret
    value: "<your client secret>"
  - name: scopes
    value: "admin:read admin:write user:read user:write"
  - name: authURL
    value: "https://accounts.google.com/o/oauth2/v2/auth"
  - name: tokenURL
    value: "https://accounts.google.com/o/oauth2/token"
  - name: redirectURL
    value: "http://myapp.com/callback"
  - name: pathFilter
    value: "^/api/admin/.*"
```

> In this configuration:

Reviewer: ```suggestion
```

> - Requests to `/api/users/*` endpoints receive tokens with limited user scopes

Reviewer: ```suggestion
```

> - Requests to `/api/admin/*` endpoints receive tokens with full administrative privileges

Reviewer: ```suggestion
```

> - This reduces security risk by preventing privilege escalation and limiting the blast radius of compromised tokens

Reviewer: ```suggestion
```
